pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go

```diff
@@ -367,6 +369,66 @@ func buildFakeServer(t *testing.T) string {
 	return filepath.Dir(serverBin)
 }
 
+func TestPythonFromBashScript(t *testing.T) {
+	curDir, err := testutil.CurDir()
+	require.NoError(t, err)
+	pythonScriptPath := filepath.Join(curDir, "testdata", "script.py")
+
+	t.Run("PythonFromBashScript", func(t *testing.T) {
+		testCaptureWrappedCommands(t, pythonScriptPath, []string{"sh", "-c"}, func(service model.Service) bool {
+			return service.Language == string(language.Python)
+		})
+	})
+	t.Run("DirectPythonScript", func(t *testing.T) {
+		testCaptureWrappedCommands(t, pythonScriptPath, nil, func(service model.Service) bool {
+			return service.Language == string(language.Python)
+		})
+	})
+}
+
+func testCaptureWrappedCommands(t *testing.T, script string, commandWrapper []string, validator func(service model.Service) bool) {
+	// Changing permissions
+	require.NoError(t, os.Chmod(script, 0755))
+
+	commandLineArgs := append(commandWrapper, script)
+	cmd := exec.Command(commandLineArgs[0], commandLineArgs[1:]...)
+	// Running the binary in the background
+	require.NoError(t, cmd.Start())
+
+	var proc *process.Process
+	var err error
+	require.EventuallyWithT(t, func(collect *assert.CollectT) {
+		proc, err = process.NewProcess(int32(cmd.Process.Pid))
+		assert.NoError(collect, err)
+	}, 10*time.Second, 100*time.Millisecond)
+
+	cmdline, err := proc.Cmdline()
+	require.NoError(t, err)
+	// If we wrap the script with `sh -c`, we can have differences between a local run and a kmt run, as for
+	// kmt `sh` is symbolic link to bash, while locally it can be a symbolic link to dash. In the dash case, we will
+	// see 2 processes `sh -c script.py` and a sub-process `python3 script.py`, while in the bash case we will see
+	// only `python3 script.py`. We need to check for the command line arguments of the process to make sure we
+	// are looking at the right process.
+	if cmdline == strings.Join(commandLineArgs, " ") && len(commandWrapper) > 0 {
+		var children []*process.Process
+		assert.EventuallyWithT(t, func(collect *assert.CollectT) {
```

Comment:
`s/assert/require`? Other there could be a panic on the `proc=children[0]` line.